### PR TITLE
doc: Documentation fixes

### PR DESCRIPTION
**Pull request by umapraseeda**
Documentation fixes
NCSDK-25854

### CODE REVIEW

**Comment by peknis** [comment]
A generic finding in all header files. Many of them start the brief for a function with "Function to do this and that." For example "Function to configure AES..." In my opinion, they should be instead like: "Configure AES..."

File: openthread/include/openthread/thread_ftd.h [hunk 1]
```diff
@@ -183,7 +183,7 @@ otError otThreadSetRouterEligible(otInstance *aInstance, bool aEligible);
  * Set the preferred Router Id.
  *
  * Upon becoming a router/leader the node attempts to use this Router Id. If the preferred Router Id is not set or if
- * it can not be used, a randomly generated router id is picked. This property can be set only when the device role is
+ * it cannot be used, a randomly generated router id is picked. This property can be set only when the device role is
```
Comment (peknis) [inline openthread/include/openthread/thread_ftd.h:186]:
```suggestion
 * it cannot be used, a randomly generated Router ID is picked. This property can be set only when the device role is
```
Not sure whether router needs to be capitalized, but it was in other instances.

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

File: nrf_modem/include/nrf_modem_gnss.h [hunk 2]
```diff
@@ -1332,7 +1332,7 @@ int32_t nrf_modem_gnss_agnss_write(void *buf, int32_t buf_len, uint16_t type);
  * @details Returns detailed information about the current GNSS A-GNSS data need. The data
  *          structure contains expiry times in seconds for different types of assistance data.
  *          This function can be called at any time. If GNSS doesn't know the current GPS
- *          system time, the expiry times can not be calculated
+ *          system time, the expiry times cannot be calculated
```
Comment (richabp) [inline nrf_modem/include/nrf_modem_gnss.h:1335]:
```suggestion
 *          system time, the expiry times cannot be calculated.
```

File: nrf_wifi/fw_if/umac_if/inc/default/fmac_structs.h [hunk 3]
```diff
@@ -445,7 +445,7 @@ struct raw_tx_stats {
 /**
  * @brief Structure to hold per device context information for the UMAC IF layer.
  *
- * This structure maintains the context information necessary for the
+ * This structure maintains the context information necessary for
  * a single instance of an FullMAC based RPU.
```
Comment (richabp) [inline nrf_wifi/fw_if/umac_if/inc/default/fmac_structs.h:449]:
```suggestion
 * a single instance of a FullMAC-based RPU.
```

File: nrf_wifi/fw_if/umac_if/inc/radio_test/fmac_structs.h [hunk 4]
```diff
@@ -27,7 +27,7 @@
 /**
  * @brief  Structure to hold per device context information for the UMAC IF layer.
  *
- * This structure maintains the context information necessary for the
+ * This structure maintains the context information necessary for
  * a single instance of an FullMAC based RPU.
```
Comment (richabp) [inline nrf_wifi/fw_if/umac_if/inc/radio_test/fmac_structs.h:31]:
```suggestion
 * a single instance of a FullMAC-based RPU.
```

File: openthread/include/openthread/thread_ftd.h [hunk 5]
```diff
@@ -180,11 +180,11 @@ bool otThreadIsRouterEligible(otInstance *aInstance);
 otError otThreadSetRouterEligible(otInstance *aInstance, bool aEligible);
 
 /**
- * Set the preferred Router Id.
+ * Set the preferred router ID.
  *
- * Upon becoming a router/leader the node attempts to use this Router Id. If the preferred Router Id is not set or if
- * it can not be used, a randomly generated router id is picked. This property can be set only when the device role is
- * either detached or disabled.
+ * Upon becoming a router or a leader the node attempts to use this router Id. If the preferred router ID is not set
```
Comment (richabp) [inline openthread/include/openthread/thread_ftd.h:185]:
```suggestion
 * Upon becoming a router or a leader, the node attempts to use this router ID. If the preferred router ID is not set
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Comment by edmont** [comment]
We don't modify the `openthread` headers directly, they are updated automatically when creating the libraries based on sources. The proper way to fix doc issues there would be upstream.

**Comment by umapraseeda** [comment]
> We don't modify the `openthread` headers directly, they are updated automatically when creating the libraries based on sources. The proper way to fix doc issues there would be upstream.

Ok. Tagging @wiba-nordic  for following this up in upstream later. I shall remove the changes to the openthread header files for now.

**Review by richabp — APPROVED** [review]

**Review by rugeGerritsen — APPROVED** [review]
Looked through the changes in ./mpsl and ./softdevice_controller

**Pull request merged**